src/ITCC.YandexSpeeckKitClient/Models/StartSessionResult.cs

```diff
@@ -0,0 +1,95 @@
+﻿// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+using System;
+using System.Net.Sockets;
+using System.Security.Authentication;
+using ITCC.YandexSpeeckKitClient.Enums;
+using ITCC.YandexSpeeckKitClient.MessageModels.StreamingMode;
+
+namespace ITCC.YandexSpeeckKitClient.Models
+{
+    /// <summary>
+    /// Start speech recognotion session response.
+    /// </summary>
+    public class StartSessionResult
+    {
+
+        public static StartSessionResult TimedOut { get; } = new StartSessionResult { TransportStatus = TransportStatus.Timeout };
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public TransportStatus TransportStatus { get; private set; }
+
+        /// <summary>
+        /// The response code.
+        /// </summary>
+        public ResponseCode ResponseCode { get; }
+
+        /// <summary>
+        /// Error message text. Included in the response if the response code is something other than 200.
+        /// </summary>
+        public string ApiErrorMessage { get; }
+
+        /// <summary>
+        /// The session ID. Specify this ID when contacting tech support.
+        /// </summary>
+        public string SessionId { get; }
+
+        /// <summary>
+        /// Contains server hello response.
+        /// </summary>
+        public string ServerHelloResponse { get; }
+
+        /// <summary>
+        /// Specifies socker error if TransportStatus = SocketError.
+        /// </summary>
+        public SocketError? SocketError { get; }
```

Comment:
If `SocketError.Success` is used everywhere in BCL socket APIs in case of success then our API becomes more consistent.
Don't think this is really a big deal though